### PR TITLE
upgrade aws-alb-ingress-controller to aws-load-balancer-controller v2

### LINKED ISSUE #193: Upgrade from aws-alb-ingress-controller to aws-load-balancer-controller
aws-alb-ingress-controller was renamed to aws-load-balancer-controller with the v2 release

It also now publishes images to ecr so we can avoid docker hub https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases

It also has some features to share elbs between ingresses 🎉  https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/914

Lets upgrade to the latest version... whist checking what happens to existing elbs ... IIRC the upgrade should be safe.

### PR DESCRIPTION
**Pull request by ettiee**
Closes #193 

- add helm chart deployment for aws-load-balancer-controller v2
- add kustomize for required CRD for aws-load-balancer-controller v2

### CODE REVIEW

**Comment by dan-slinky-ckpd** [comment]
can you rebase from master 🙏

File: UPGRADING.md [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 ## 1.17 -> 1.18
 
+[#195](https://github.com/cookpad/terraform-aws-eks/pull/195) upgrades `aws-alb-ingress-controller` to [`aws-load-balancer-controller`](https://github.com/cookpad/terraform-aws-eks/pull/195) (the project was renamed with the v2 release). Check the [upgrade guide for this project](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.0/guide/upgrade/migrate_v1_v2/) if you are using `aws-alb-ingress-controller`, you may need to update your ingress definitions.
```
Comment (dan-slinky-ckpd) [inline UPGRADING.md:10]: should be 18->19 ?

File: hack/generate_addons.sh [hunk 2]
```diff
@@ -19,10 +20,11 @@ kustomize_build() {
 
 helm_template eks aws-node-termination-handler 0.13.3
 helm_template autoscaler cluster-autoscaler 1.0.4 -chart
-helm_template nvdp nvidia-device-plugin 0.8.2
+helm_template nvdp nvidia-device-plugin 0.7.0
```
Comment (dan-slinky-ckpd) [inline hack/generate_addons.sh:23]: why downgrading nvidia-device-plugin here? rebase from master.

File: hack/generate_addons.sh [hunk 3]
```diff
@@ -19,10 +20,11 @@ kustomize_build() {
...
+helm_template eks aws-load-balancer-controller 1.1.0
```
Comment (dan-slinky-ckpd) [inline hack/generate_addons.sh:24]: why `1.1.0` and not the current `1.1.5`? Afterall, that is the version defined in the gitRef we're getting the CRDs from below.

File: hack/generate_addons.sh [hunk 4]
```diff
@@ -19,10 +20,11 @@ kustomize_build() {
...
 
 curl -o $ADDONS_DIR/kustomize/overlays/metrics-server/resources/metrics-server.yaml -L https://github.com/kubernetes-sigs/metrics-server/releases/download/v0.3.7/components.yaml
 kustomize_build metrics-server
 curl -o $ADDONS_DIR/kustomize/overlays/aws-ebs-csi-driver/resources/crd_snapshotter.yaml -L https://raw.githubusercontent.com/kubernetes-sigs/aws-ebs-csi-driver/v0.7.0/deploy/kubernetes/cluster/crd_snapshotter.yaml
 kustomize_build aws-ebs-csi-driver
-kustomize_build aws-alb-ingress-controller
+kustomize_build aws-load-balancer-controller-crds
```
Comment (dan-slinky-ckpd) [inline hack/generate_addons.sh:30]: We're getting the manifests from the helm chart, but the CRDs via a kustomize base in the repo repository. Can we not just references the bases from https://github.com/kubernetes-sigs/aws-load-balancer-controller/tree/main/config

File: modules/cluster/addons/kustomize/overlays/aws-load-balancer-controller-crds/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
```
Comment (dan-slinky-ckpd) [inline modules/cluster/addons/kustomize/overlays/aws-load-balancer-controller-crds/kustomization.yaml:3]: `bases` is deprecated.

```suggestion
resources:
```

File: modules/cluster/README.md [hunk 6]
```diff
@@ -94,7 +94,7 @@ specify the arn of an existing key by setting `kms_cmk_arn`
 | [Prometheus Node Exporter](https://github.com/prometheus/node_exporter) | `prometheus_node_exporter` | ❌ disabled ||
 | [Kubernetes Metrics Server](https://github.com/kubernetes-sigs/metrics-server) | `metrics_server` | ❌ disabled ||
 | [Amazon Elastic Block Store (EBS) CSI driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/) | `aws_ebs_csi_driver` | ❌ disabled | `aws_ebs_csi_driver_iam_role_arn` |
-| [AWS ALB Ingress Controller](https://github.com/kubernetes-sigs/aws-alb-ingress-controller) | `aws_alb_ingress_controller` | ❌ disabled | `aws_alb_ingress_controller_iam_role_arn` |
+| [AWS Load Balancer Controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller) | `aws_load_balancer` | ❌ disabled | `aws_load_balancer_iam_role_arn` |
```
Comment (dan-slinky-ckpd) [inline modules/cluster/README.md:97]:
```suggestion
| [AWS Load Balancer Controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller) | `aws_load_balancer_controller` | ❌ disabled | `aws_load_balancer_iam_role_arn` |
```

File: UPGRADING.md (hunk 1 above)
Comment (ettiee) [reply UPGRADING.md:10]: ah yes ta

File: hack/generate_addons.sh (hunk 4 above)
Comment (ettiee) [reply hack/generate_addons.sh]: yeah that makes sense

**Comment by ettiee** [comment]
@dan-slinky-ckpd thanks for your review 🙏 
Pushed updates for your comments, `nvcr.io` was down today but it's back alive so I've kick started the test suite again

File: modules/cluster/addons/helm/aws-load-balancer-controller.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+tolerations:
+- key: CriticalAddonsOnly
+  operator: Exists
+image:
+  repository: 602401143452.dkr.ecr.${aws_region}.amazonaws.com/amazon/aws-load-balancer-controller
+  tag: v2.1.0
```
Comment (dan-slinky-ckpd) [inline modules/cluster/addons/helm/aws-load-balancer-controller.yaml:6]:
```suggestion
  tag: v2.1.3
```

This will then match `app.kubernetes.io/version`

**Review by dan-slinky-ckpd — APPROVED** [review]
LGTM. Suggested changed to the image tag.

**Pull request merged**